avm/src/daemon/ctrl.rs

```diff
@@ -76,9 +76,32 @@ impl LogRendezvousHash {
     &mut self.sorted_hashes
   }
 
+  pub fn _get_primary_node_id(&self, key: &str) -> &str {
+    self._get_assigned_nodes_id(key)[0]
+  }
+
+  pub fn _get_assigned_nodes_id(&self, key: &str) -> Vec<&str> {
+    let top = 3; // Number of replicas
+    if self.sorted_hashes.len() == 1 {
+      // If only one node, nothing to do
+      return self
+        .sorted_hashes
+        .iter()
+        .map(|hash| &hash.id as &str)
+        .collect::<Vec<&str>>();
+    }
+    let mut idx = self._get_idx_for_key(key);
+    let mut ids: Vec<&str> = Vec::new();
+    for _ in 0..top {
+      ids.push(&self.sorted_hashes[idx].id);
+      idx = (idx + 1) % self.sorted_hashes.len();
+    }
+    ids
```

Comment:
This will contain a node twice if the cluster size is 2.

While short-circuiting for a single node is a decent idea, it's also saving compute time on the cheapest path, anyways, so I don't know if we need to worry about that special casing, maybe we can replace this with something like:

`pub fn get_assigned_nodes_id(&self, key: &str) -> Vec<&str> {
  let top = cmp::min(3, self.sorted_hashes.len());
  let mut idx = self.get_idx_for_key(key);
  let mut ids = Vec::new();
  for _ in 0..top {
    ids.push(&self.sorted_hashes[idx].id);
    idx = (idx + 1) % self.sorted_hashes.len();
  }
  ids
}
```

Reply:
Oh yes, my bad that I did not handle that case. This looks cleaner!